src/main/java/org/springframework/data/repository/support/DefaultRepositoryInvokerFactory.java

```diff
@@ -93,10 +93,7 @@ private RepositoryInvoker prepareInvokers(Class<?> domainType) {
 	@SuppressWarnings("unchecked")
 	protected RepositoryInvoker createInvoker(RepositoryInformation information, Object repository) {
 
-		if (repository instanceof PagingAndSortingRepository) {
```

Comment:
Don't we need the `PagingAndSortingRepositoryInvoker` anymore?

Reply:
That is the main reason why I wanted you to take look at this.

It is not technically necessary, but I guess this comes with a penalty hit. 
No idea how big and relevant that is.

The alternative would be to provide Invokers for different combinations of `CrudRepositories` and `PagingAndSortingRepository` and possibly others.
Smells of code explosion.